app/webpacker/scripts/form-check-leave.js

```diff
@@ -0,0 +1,33 @@
+function FormCheckLeave($module) {
+  this.$module = $module;
+  this.$data = "";
+}
+
+FormCheckLeave.prototype.data = function() {
+  const $form = this.$module;
+  return Array.from(new FormData($form), e =>
+    e.map(encodeURIComponent).join("=")
+  ).join("&");
+};
+
+FormCheckLeave.prototype.init = function() {
+  const $form = this.$module;
+
+  if (!$form) return;
+  const $data = this.data();
+  this.action($form, $data);
+};
+
+FormCheckLeave.prototype.action = function($form, $data) {
+  const $changes = this.data();
+  $form.addEventListener("submit", () => (window.onbeforeunload = null));
+
+  window.onbeforeunload = function() {
```

Comment:
Do we need to pass in event here?
`window.onbeforeunload = function(event) {`

Reply:
Do we need some comparison of original data and new data here?

Reply:
`event` is always present but I will add it in for clarification. 

Have a fix for the comparison, not sure when that disappeared.